graph/path/bellman_ford_moore_test.go

```diff
@@ -41,7 +39,7 @@ func TestBellmanFordFrom(t *testing.T) {
 			t.Errorf("%q: unexpected weight from To: got:%f want:%f",
 				test.Name, weight, test.Weight)
 		}
-		if weight := pt.WeightTo(test.Query.To().ID()); weight != test.Weight {
+		if weight := pt.WeightTo(test.Query.To().ID()); !math.IsInf(test.Weight, -1) && weight != test.Weight {
```

Comment:
Negative infinite path weights fall out of finding shortest (lowest cost) paths in a graph with cycles with single traversal costs that are negative; repeated traversal of the cycle reduces the total cost of the path by the single traversal cost each time. Since there is no constraint to limit the number of traversals you end up with -Inf.

Reply:
Is there a comment somewhere saying this? Unless you think the reader should know this.

Reply:
The reader should know this, though there are examples that demonstrates how to find them: https://godoc.org/gonum.org/v1/gonum/graph/path#example-BellmanFordFrom--Negativecycles and https://godoc.org/gonum.org/v1/gonum/graph/path#example-FloydWarshall--Negativecycles.